modules/documents/client/directives/documents.confirm.unpublish.directive.js

```diff
@@ -0,0 +1,302 @@
+            function checkFileForCollections(file, selectedFiles) {
+              var promises = [];
+              _.forEach(file.collections, function(collection) {
+                promises.push(new Promise (function (resolve, reject) {
+                  CollectionModel.getModel(collection.id)
+                    .then(function(result) {
+                      // Is this file in the *main* documents of the collection?
+                      var found = _.find(result.mainDocuments, function(doc) {
+                        return doc.document.id == file.id;
+                      })
+                      if (found) {
+                        // If all main documents of this collection are trying to be unpublished...
+                        if (assessCollection(selectedFiles, result.mainDocuments)) {
+                          // Restrict this document (so it won't be unpublished).
+                          file.restrictingCollections.push(result.displayName);
+                        }
+                      }
+                      return resolve(result);
+                    })
+                    .catch(function (err) {
+                      return reject(err);
+                    })
+                }))
+              })
+              return Promise.all(promises);
+            }
+
+            // This helper function determines if unpublishing all selected documents would leave the collection without a published main document.
+            function assessCollection(selectedDocs, collectionDocs) {
+              var sDocs = _.map(selectedDocs, function(s) {
+                return s.id;
+              })
+              // Only consider published documents in the collection
+              var cDocs = _.filter(collectionDocs, function(c) {
+                return c.document.isPublished;
+              });
+              cDocs = _.map(cDocs, function(c) {
+                return c.document.id;
+              });
+              // Remove all checked items from the collection's published main docs. Is there anything left?
+              cDocs = _.difference(cDocs, sDocs);
+              return cDocs.length == 0;
+            }
+
+            function updateText() {
+              var folderCnt = self.folders.length;
+              var fileCnt = self.files.length;
+              var unpublishableFolderCnt = self.unpublishableFolders.length;
+              var unpublishableFileCnt = self.unpublishableFiles.length;
+              self.allBlocked = false;
+              self.hasBlockedContent = false;
+
+              if (unpublishableFolderCnt > 0 || unpublishableFileCnt > 0) {
+                self.confirmText = "Are you sure you want to unpublish the following file(s) and/or folder(s)?";
+                if (folderCnt > unpublishableFolderCnt || fileCnt > unpublishableFileCnt ) {
+                  self.hasBlockedContent = true;
+                  self.bannerText = " One or more of the following files or folders cannot be unpublished.";
```

Comment:
Nit.: heading spaces in message string (in next ones as well).

Reply:
There were changes to the .scss; these were necessary but may no longer be. I will check, and remove if possible.